### ISSUE TITLE
Step-by-step bear guide

### ISSUE THREAD

**Issue by Makman2** [coala-analyzer/coala#561]
The *writing-bears* guide should be improved. Some notes here and there, but the major aspect I have in mind:
A step-by-step guide that implements an actual bear of our coalib (in a simplified manner), so the user sees how to operate on files and do something meaningful. In my mind is the indentation bear, that should be fairly simple for one language (So we don't take the general LanguageDefinition feature in account or whatever).

One more idea:
- A little paragraph about global bears (as demonstration: A bear that calculates the MD5 hash - or any other hash-value - from all files)

**Comment by sils1297** [comment]
I think we should take that LanguageDefinition into account, however it will be realized, because this is one of the key thoughts on writing bears.

Also we should make an example about dependent bears.

**Comment by Makman2** [comment]
Maybe we should extend the simple bear with the LanguageDefinition feature. First use hardcoded values, and then introduce the bear-writer into our cool language mechanics :)

**Comment by sils1297** [comment]
sure something like that.

**Comment by sils1297** [comment]
lets just first figure out how to make language mechanics cool ;)

**Comment by sils1297** [comment]
I have a good example: PyLintBear - truly simple, truly useful.

**Comment by Makman2** [comment]
Oh yeah right, this would be a perfect example :)

**Comment by sils1297** [comment]
blocked by the new bear design

**Comment by Makman2** [comment]
You think we should wait for the new design? We could already make the guide and port it later^^

**Comment by sils1297** [comment]
we could do that but we need to make sure that the new tutorial is well
ported then, not half heartedly adapted.

2015-07-16 18:24 GMT+02:00 Makman2 <notifications@github.com>:

**Comment by Makman2** [comment]
That's an argument, keep it for 0.3 alpha ^^

**Comment by Makman2** [comment]
Moving this to 0.4, we don't get the new bear-ecosystem done for 0.3.

**Comment by sils1297** [comment]
Not milestone relevant and we don't know when the new bear design is goig to happen.

**Comment by Makman2** [comment]
We have somehow a guide, now I don't know how to improve it...
@sils1297 closing this issue?

**Comment by sils1297** [comment]
@Makman2 no we need to revamp the bear writing tutorial. That's really important.

**Comment by AbdealiJK** [comment]
Do we need this still ? We have a pretty decent Writing_Bears.rst and now Linter_Bear.rst and Self_Correcting_bears.rst have been added.

PyLintBear is now a LinterBear, so that's not a good example to use
Maybe we could use LineCountingBear ?

Better description of what needs to done is required. Possibly split this into smaller newcomer tasks would be a good idea ?

**Comment by sils1297** [comment]
I do think we should improve our original writing bears tutorial.

**Comment by Adrianzatreanu** [comment]
Working on this on the new docs, using the new linter API

**Comment by Adrianzatreanu** [comment]
This issue was moved to coala-analyzer/documentation#58
